configure.ac

```diff
@@ -0,0 +1,65 @@
+AC_INIT([chaosnet-bridge], 0.0)
+AC_CONFIG_AUX_DIR([build-aux])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+
+AC_CONFIG_SRCDIR([cbridge.c])
+
+AC_ARG_ENABLE([etherp],
+    AS_HELP_STRING([--enable-etherp], [Enable Chaos-over-Ether code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-etherp])])],
+    [enable_etherp=yes])
+enable_etherp_val=0
+if test "x$enable_etherp" = "xyes"; then
+    enable_etherp_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_ETHERP], [$enable_etherp_val], [])
+AM_CONDITIONAL([ENABLE_ETHERP], [test x$enable_etherp = xyes])
+
+AC_ARG_ENABLE([tls],
+    AS_HELP_STRING([--enable-tls], [Enable Chaos-over-TLS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-tls])])],
+    [enable_tls=yes])
+enable_tls_val=0
+if test "x$enable_tls" = "xyes"; then
+    enable_tls_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_TLS], [$enable_tls_val], [])
+AM_CONDITIONAL([ENABLE_TLS], [test x$enable_tls = xyes])
+
+AC_ARG_ENABLE([ip],
+    AS_HELP_STRING([--enable-ip], [Enable Chaos-over-IP code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-ip])])],
+    [enable_ip=yes])
+enable_ip_val=0
+if test "x$enable_ip" = "xyes"; then
+    enable_ip_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_IP], [$enable_ip_val], [])
+AM_CONDITIONAL([ENABLE_IP], [test x$enable_ip = xyes])
+
+# CHAOS_DNS // enable the DNS code (forwarder and internal host/addr parser)
+AC_ARG_ENABLE([dns],
+    AS_HELP_STRING([--enable-dns], [Enable DNS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-dns])])],
+    [enable_dns=yes])
+enable_dns_val=0
+if test "x$enable_enable_dns" = "xyes"; then
+    enable_dns_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_DNS], [$enable_dns_val], [])
+AM_CONDITIONAL([ENABLE_DNS], [test x$enable_dns = xyes])
+
+#	CHAOS_DNS_SERVER
```

Comment:
@bictorv  Do you think it makes sense to set these values via configure? Or should these be configurable via the config?

Reply:
What values do you mean? I think it does make sense to be able to un-define e.g. CHAOS_TLS and remove chtls.o and the libs needed for that - and similar for the other "modules".

Reply:
That is already possible, `—disable-tls` for example.  What I was referring to was the CHAOS_DNS_SERVER macro, and its other friend that I forgot the name of.

Reply:
Ah. No, I think the server address (Psilo) should require manual work to change, and also the address domain CH-ADDR.NET.